jvm/src/test/scala/tastyquery/PositionSuite.scala

```diff
@@ -0,0 +1,290 @@
+  }
+
+  test("apply") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"EtaExpansion")
+    assertEquals(collectCode[Apply](tree, code), List("f(0)", "takesFunction(intMethod)", "intMethod"))
+  }
+
+  /** Control structures */
+
+  test("if") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"If")
+    assertEquals(collectCode[If](tree, code), List("if (x < 0) -x else x"))
+  }
+
+  test("while") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"While")
+    assertEquals(
+      collectCode[While](tree, code),
+      List("""while (true) {
+             |      ()
+             |    }""".stripMargin)
+    )
+  }
+
+  test("match") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"Match")
+    assertEquals(
+      collectCode[Match](tree, code),
+      List("""x match {
+             |    case 0 => 0
+             |    case 1 | -1 | 2 => x + 1
+             |    case 7 if x == 7 => x - 1
+             |    case 3 | 4 | 5 if x < 5 => 0
+             |    case _ if (x % 2 == 0) => x / 2
+             |    case _ => -x
+             |  }""".stripMargin)
+    )
+  }
+
+  test("case-def") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"Match")
+    assertEquals(collectCode[CaseDef](tree, code), Nil)
+  }
+
+  test("bind") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"Bind")
+    assertEquals(collectCode[Bind](tree, code), List("t @ y", "y", "s: String", "k @ Some(_)"))
+  }
+
+  /** Functions */
+
+  test("def-def") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"Function")
+    assertEquals(collectCode[DefDef](tree, code), List("(x: Int) => x + 1", "() => ()"))
+  }
+
+  test("def-def-nested") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"NestedMethod")
+    assertEquals(
+      collectCode[DefDef](tree, code),
+      List(
+        """def outerMethod: Unit = {
+          |    def innerMethod: Unit = ()
+          |  }""".stripMargin,
+        "def innerMethod: Unit = ()"
+      )
+    )
+  }
+
+  test("named-arg") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"NamedArgument")
+    assertEquals(collectCode[NamedArg](tree, code), List("second = 1"))
+  }
+
+  test("block") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"Block")
+    assertEquals(
+      collectCode[Block](tree, code),
+      List("""{
+             |    val a = 1
+             |    val b = 2
+             |    ()
+             |  }""".stripMargin)
+    )
+  }
+
+  test("lambda") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"Function")
+    assertEquals(
+      collectCode[Lambda](tree, code),
+      // NoSpan is the expected behavior
+      Nil
+    )
+  }
+
+  test("return") {
+    val (tree, code) = unpickleWithCode(simple_trees / tname"Return")
+    assertEquals(collectCode[Return](tree, code), List("return 1"))
+    assertEquals(collectCode[Ident](tree, code), List("1"))
```

Comment:
I'm not sure what this is, but if it is the Ident of the Return then the Id should have no span